cypress/support/page-objects/filters.page.js

```diff
@@ -0,0 +1,119 @@
+// cypress/support/page-objects/filters.page.js
+import BasePage from './base-page'
+
+class FiltersPage extends BasePage {
+    elements = {
+        resultsSummary: () => cy.get('[data-v-1549eec6].search-results__summary'),
+        searchResults: () => cy.get('[data-v-1549eec6].search-results__list'),
+        searchResultsSummary: () => cy.get('[data-v-1549eec6].search-results__summary'),
+        categoryPill: () => cy.get('[data-v-ed449272].removable-pill.search-results__summary-category-pill'),
+        categoryWrapper: () => cy.get('[data-v-1549eec6].search-results__summary-category-wrapper'),
+        removePillButton: () => cy.get('[data-v-ed449272].removable-pill__remove-icon.material-icons'),
+        currencySelect: () =>  cy.get('[data-v-baa43182].currency-dropdown__select').eq(0).select('GBP', { force: true }),
+        storeCountrySelect: () => cy.get('.country-dropdown__select'),
+        shipCountrySelect: () => cy.get('.country-dropdown__select-padding-flag'),
+        babyCategory: () => cy.contains('Baby'),
+        searchResultsCount: () => cy.get('.search-results__result-heading'),
+        productCard: () => cy.get('.product-search-card.search-result-card'),
+    }
+
+    selectFirstProduct() {
+        this.elements.productCard().first().click();
+        return this;
+    }
+
+    verifyResultsCountGreaterThanZero() {
+        this.elements.searchResultsCount().should('have.length.greaterThan', 0);
+        return this;
+    }
+
+    selectBabyCategory() {
+        this.elements.babyCategory().click()
+        return this
+    }
+
+    selectStoreCountry(country) {
+        this.elements.storeCountrySelect()
+            .eq(0)
+            .should('exist')
+            .select(country, { force: true });
+        return this;
+    }
+
+    verifyStoreCountrySelection(country) {
+        this.elements.storeCountrySelect().eq(0).should('have.value', 'GB');
+    }
+
+    selectShippingCountry(country) {
+        // Ensure the dropdown exists and is visible
+        this.elements.shipCountrySelect().should('exist').should('be.visible')
```

Comment:
I notice a few places where you check for existence and visibility before actioning. Can you talk us through the rationale there please.

Reply:
Yes, I perform existence and visibility checks in certain places for specific reasons:

1. Consistent checks for existence and visibility before actions provide clear debugging benefits:
Better Error Messages:
- Without checks: cy.click() fails with a generic "element not found" error
- With checks: We know specifically if the element:
           - Doesn't exist in DOM (should('exist') failed)
           - Exists but isn't visible (should('be.visible') failed)

2. For dropdown elements (like in selectStoreCountry), verified using: 
**.should('exist').should('be.visible')**
Because dropdowns can be conditionally rendered or hidden based on other selections.

3. For simple elements (like in selectBabyCategory), went with: **.click()**
Because these are static elements that are always present in the DOM